### PR TITLE
Introduced Keycloak service

### PR DESCRIPTION
**Pull request by gastaldi**
- [ ] Have you created an [issue](https://github.com/redhat-kontinuity/catapult/issues) and used it in the commit message?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/redhat-kontinuity/catapult/pulls) for the same issue?
- [X] Have you built the project locally prior to submission with `mvn clean install`?

-----

### CODE REVIEW

**Comment by gastaldi** [comment]
Once this PR is merged, Catapult will require the `KEYCLOAK_URL` and `KEYCLOAK_REALM` environment variables to deploy successfully

File: services/keycloak-service-impl/src/test/java/org/kontinuity/catapult/service/keycloak/impl/KeycloakServiceImplTest.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    public void testBuildUrl() {
+        String url = KeycloakServiceImpl.buildURL("http://sso.prod-preview.openshift.io", "fabric8", "github");
+        Assert.assertEquals("http://sso.prod-preview.openshift.io/auth/realms/fabric8/broker/github/token", url);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testInvalidTokenOpenshift() {
+        String token = "token";
+        KeycloakServiceImpl service = new KeycloakServiceImpl("http://sso.prod-preview.openshift.io", "fabric8");
+        System.out.println(service.getOpenshiftV3Token(token));
```
Comment (edewit) [inline services/keycloak-service-impl/src/test/java/org/kontinuity/catapult/service/keycloak/impl/KeycloakServiceImplTest.java]: oops you left this by accident

Comment (gastaldi) [inline services/keycloak-service-impl/src/test/java/org/kontinuity/catapult/service/keycloak/impl/KeycloakServiceImplTest.java]: Not really, just wanted to display the token :) Hopefully the System.out will not be called in this test ;)

**Comment by ALRubinger** [comment]
Addresses #181

**Comment by ALRubinger** [comment]
cbd21ef9700a4d986444efcd4b8367d01d4cd08e

**Pull request closed without merging**